### PR TITLE
QMAPS-1761 don't restore last location on load in iframe mode

### PR DESCRIPTION
**Pull request by xem**
## Description
- don't restore last location on load in iframe mode (&no_ui=1)
- don't save last location in iframe mode

can we test it on dev?

## Why
To avoid thys flyTo when loading the IA
fly

### CODE REVIEW

File: src/panel/poi/PoiPanel.jsx [hunk 1]
```diff
@@ -192,7 +192,9 @@ export default class PoiPanel extends React.Component {
     const uri = `/places/${params}`;
 
     Telemetry.add(Telemetry.POI_BACKTOLIST);
-    fire('restore_location');
+    if (!window.no_ui) {
+      fire('restore_location');
+    }
```
Comment (amatissart) [inline src/panel/poi/PoiPanel.jsx:197]: I am not sure to understand how this change is supposed to solve the problem you are describing.  
This function `backToList` is used only when clicking on the button at the top of the PoiPanel after an item has been selected from a list. Or am I missing something ?

**Comment by amatissart** [comment]
> since pan/zoom is not available in this mode, the app can't save the last visited position,

The map position will be saved anyway if the query triggers a map move (e.g "restaurant à Nice" which requires the map to be centered on the detected zone)

File: src/panel/poi/PoiPanel.jsx (hunk 1 above)
Comment (xem) [reply src/panel/poi/PoiPanel.jsx:197]: Ah, let me check, there may be a mistake as there are 2 different "getLastLocation()" functions in the project =_=

**Review by bbecquet — APPROVED** [review]

File: src/adapters/scene.js [hunk 2]
```diff
@@ -208,13 +208,15 @@ Scene.prototype.initMapBox = async function(locationHash) {
...
-      window.app.updateHash(this.getLocationHash());
-      fire('map_moveend');
-    });
+    if (!window.no_ui) {
+      this.mb.on('moveend', () => {
+        const { lng, lat } = this.mb.getCenter();
+        const zoom = this.mb.getZoom();
+        store.setLastLocation({ lng, lat, zoom });
+        window.app.updateHash(this.getLocationHash());
+        fire('map_moveend');
```
Comment (amatissart) [inline src/adapters/scene.js:217]: This `map_moveend` event is used by the `CategoryPanel` to trigger the search for POIs in the current zone after the intention has been derived from the query and display the markers.  
So it should still be fired. As is no marker would be visible on the map.

**Review by amatissart — CHANGES_REQUESTED** [review]

**Review by amatissart — APPROVED** [review]

**Pull request merged**